File: bindings/python/examples/how_tos/accounts_and_addresses/list_outputs.py
import os

from dotenv import load_dotenv

from iota_sdk import Wallet, WalletOptions

# In this example we will get outputs stored in the wallet

# This example uses secrets in environment variables for simplicity which
# should not be done in production.
load_dotenv()

wallet = Wallet(WalletOptions(storage_path=os.environ.get('WALLET_DB_PATH')))
wallet.sync()

# All outputs stored in the wallet
outputs = wallet.outputs()

# Print all output ids
print('Output ids:')
for output in outputs:
    print(output.output_id)

# All unspent outputs stored in the wallet
outputs = wallet.unspent_outputs()

# Print all unspent output ids
print('Unspent output ids:')
for output in outputs:
    print(output.output_id)

File: bindings/nodejs/examples/how_tos/accounts_and_addresses/list-outputs.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, initLogger } from '@iota/sdk';

// This example uses secrets in environment variables for simplicity which should not be done in production.
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/accounts_and_addresses/list-outputs.ts

// This example lists all outputs in the wallet.
async function run() {
    initLogger();
    for (const envVar of ['WALLET_DB_PATH']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }
    try {
        const wallet = await Wallet.create({
            storagePath: process.env.WALLET_DB_PATH,
        });

        await wallet.sync();

        const outputs = await wallet.outputs();

        console.log('Output ids:');
        for (const output of outputs) console.log(output.outputId);

        const unspentOutputs = await wallet.unspentOutputs();

        console.log('Unspent output ids:');
        for (const output of unspentOutputs) console.log(output.outputId);
    } catch (error) {
        console.error('Error: ', error);
    }
}

void run().then(() => process.exit());
